src/daft-sql/src/planner.rs

```diff
@@ -480,50 +493,71 @@ impl SQLPlanner {
                 })
             }
         };
+
         if root == current_relation.get_name() {
-            let column = idents.next().unwrap();
-            let column_name = ident_to_str(column);
-            let current_schema = current_relation.inner.schema();
-            let f = current_schema.get_field(&column_name).ok();
+            if idents.len() == 0 {
+                return Ok(current_relation
+                    .inner
+                    .schema()
+                    .fields
+                    .keys()
+                    .map(|f| col(f.clone()))
+                    .collect());
+            }
+
+            // If duplicate columns are present in the schema, it adds the table name as a prefix. (df.column_name)
+            // So we first check if the prefixed column name is present in the schema.
+            let current_schema = self.relation_opt().unwrap().inner.schema();
+            let f = current_schema.get_field(&ident_str).ok();
             if let Some(field) = f {
                 Ok(vec![col(field.name.clone())])
+            // If it's not, we also need to check if the column name is present in the current schema.
+            // This is to handle aliased tables. (df1 as a join df2 as b where b.column_name)
+            } else if let Some(next_ident) = idents.next() {
+                let column_name = ident_to_str(next_ident);
+
+                let f = current_schema.get_field(&column_name).ok();
+                if let Some(field) = f {
+                    Ok(vec![col(field.name.clone())])
+                } else {
+                    column_not_found_err!(&column_name, &current_relation.get_name());
+                }
             } else {
-                column_not_found_err!(&column_name, &current_relation.get_name());
+                column_not_found_err!(&ident_str, &current_relation.get_name());
             }
         } else {
             table_not_found_err!(root);
         }
     }
 
     fn select_item_to_expr(&self, item: &SelectItem) -> SQLPlannerResult<Vec<ExprRef>> {
```

Comment:
To confirm, this converts a select item to an unresolved expression, which should then be resolved in the logical plan?